src/bp-activity/bp-activity-notifications.php

```diff
@@ -246,6 +246,18 @@ function bp_activity_update_reply_add_notification( $activity, $comment_id, $com
 		}
 	}
 
+	// Try to find mentions.
+	$comment_activity = new BP_Activity_Activity( $comment_id );
+	$usernames = bp_activity_find_mentions( $comment_activity->content );
+
+	error_log( print_r( $usernames, true ) );
+	// Bail out the replied notification because he will recieve a "mentioned you" notification.
+	if ( ! empty( $usernames ) ) {
+		if ( isset( $usernames[ $activity->user_id ] ) ) {
+			return;
+		}
+	}
```

Comment:
```suggestion
	if ( ! empty( $usernames ) && array_key_exists( $activity->user_id, $usernames ) ) {
		if ( true === bb_is_notification_enabled( $activity->user_id, 'bb_new_mention' ) ) {
			return;
		}
	}
```
@rajinsharwar2 We should check if mentions are enabled for the user. If they are, return the mention notification. If not, trigger the reply notification instead.